src/java/org/apache/cassandra/transport/CQLMessageHandler.java

```diff
@@ -518,10 +533,23 @@ protected boolean processFirstFrameOfLargeMessage(IntactFrame frame, Limit endpo
             // max CQL message size defaults to 256mb, so should be safe to downcast
             int messageSize = Ints.checkedCast(header.bodySizeInBytes);
             receivedBytes += buf.remaining();
+
+            if (authMessageTooBig(messageSize))
+            {
+                // we raise a fatal error and close the connection,
+                // so it does not make sense to continue frames processing
+                ClientMetrics.instance.markRequestDiscarded();
+                return false;
+            }
             
             LargeMessage largeMessage = new LargeMessage(header);
-
-            if (throwOnOverload)
+            if (messageSize > MAX_CQL_MESSAGE_SIZE)
```

Comment:
totally agree, actually I have come to the same conclusion as well and I am already working now on such change

Reply:
there is one more thing to fix actually - handleErrorAndRelease invocation is not correct in my case onComplete, because there is no capacity to release it back, so it should be handleError